### PR TITLE
Type hints

### PR DESCRIPTION
**Pull request by dbrgn**
What do you think about those? I think it improves legibility and tooling a lot, I had to guess the type of the arguments because there was no documentation (I first assumed hex string, then bytes, both of which were wrong).

### CODE REVIEW

**Comment by lgrahl** [comment]
Sure, they'll be a great addition. You should rebase them onto the `develop` branch or wait until #21 has been merged. (Btw., although documentation isn't bad in general it's mostly missing for the callback server API.)

**Comment by dbrgn** [comment]
- Rebased against develop
- Set up conditional `typing` dependency
- Set up type checking with mypy

**Comment by lgrahl** [comment]
@dbrgn Can you rebase again? `requirements-dev.txt` shouldn't be necessary anymore (see new `setup.py`)

**Comment by dbrgn** [comment]
@lgrahl done

File: setup.py
Comment (lgrahl) [inline setup.py:45]: Any reason why this exact version? :)

**Review by lgrahl — CHANGES_REQUESTED** [review]
Looks good so far. Can you add usage of mypy to the readme?

File: setup.py
Comment (dbrgn) [reply setup.py:45]: It's just the current version. But pinning it avoids breaking of the testsuite if mypy pushes a new release with additional checks. It's probably better to simply bump the version from time to time.

Comment (lgrahl) [reply setup.py:45]: It's a bit weird that we use non-pinned dependencies for all the other tools but fair enough, I get your point. :)

**Review by lgrahl — APPROVED** [review]

**Pull request merged**
